// @uni/toast/lib/baidu-smartprogram/showToast.js
"use strict";

exports.__esModule = true;
exports.show = exports.default = void 0;
var _index = require("../utils/index");
var show = (0, _index.normalize)(function (options) {
  var type = options.type,
    content = options.content,
    duration = options.duration,
    _success = options.success,
    _fail = options.fail,
    _complete = options.complete;
  var iconMap = {
    success: 'success',
    loading: 'loading',
    none: 'none'
  };
  if (type === 'fail') {
    swan.showToast({
      image: 'https://gw.alicdn.com/imgextra/i2/O1CN01XzxlKk1GL8IyXoqEa_!!6000000000605-2-tps-200-200.png',
      title: content,
      duration: duration,
      success: function success() {
        _success && _success();
      },
      fail: function fail(res) {
        _fail && _fail(res);
      },
      complete: function complete(res) {
        _complete && _complete(res);
      }
    });
  } else {
    swan.showToast({
      icon: iconMap[type] || 'none',
      title: content,
      duration: duration,
      success: function success() {
        _success && _success();
      },
      fail: function fail(res) {
        _fail && _fail(res);
      },
      complete: function complete(res) {
        _complete && _complete(res);
      }
    });
  }
});
exports.show = show;
var _default = show;
exports.default = _default;
