// @uni/toast/lib/utils/index.js
"use strict";

exports.__esModule = true;
exports.SHORT_DELAY = exports.LONG_DELAY = void 0;
exports.normalize = normalize;
exports.styleOptions = styleOptions;
var _promisify = require("../_utils/promisify");
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
var LONG_DELAY = 3500; // 3.5 seconds
exports.LONG_DELAY = LONG_DELAY;
var SHORT_DELAY = 2000; // 2 seconds
exports.SHORT_DELAY = SHORT_DELAY;
function styleOptions(options) {
  var DEFAULT_REQUEST_OPTIONS = {
    content: '',
    type: 'none',
    duration: SHORT_DELAY
  };
  if (typeof options === 'string') {
    return _extends({}, DEFAULT_REQUEST_OPTIONS, {
      content: options
    });
  } else {
    return _extends({}, DEFAULT_REQUEST_OPTIONS, options);
  }
}
function normalize(api) {
  return function (options) {
    var afterOptions = styleOptions(options);
    return (0, _promisify.promisify)(api)(afterOptions);
  };
}
